File: Demo-ObjC/Demo-ObjC/ViewController.m
//
//  ViewController.m
//  Demo-ObjC
//
//  Created by CHEN Xian’an on 1/17/16.
//  Copyright © 2016 lazyapps. All rights reserved.
//

#import "ViewController.h"
@import MenuItemKit;

@interface ViewController ()

@property (strong, nonatomic) IBOutlet UIButton *button;

@end

@implementation ViewController

- (IBAction)tapButton:(id)sender
{
  UIMenuController *controller = [UIMenuController sharedMenuController];
  __weak typeof(self) _self = self;
  UIMenuItem *textItem = [[UIMenuItem alloc] mik_initWithTitle:@"Text" action:^(UIMenuItem *item) {
    [_self showAlertWithTitle:@"Text Item tapped"];
  }];
  
  UIImage *image = [UIImage imageNamed:@"Image"];
  UIMenuItem *imageItem = [[UIMenuItem alloc] mik_initWithTitle:@"Image" image:image action:^(UIMenuItem *item) {
    [_self showAlertWithTitle:@"Image Item Tapped"];
  }];

  UIImage *colorImage = [UIImage imageNamed:@"ColorImage"];
  UIMenuItem *colorImageItem = [[UIMenuItem alloc] mik_initWithTitle:@"Color Image" image:colorImage action:^(UIMenuItem *item) {
    [_self showAlertWithTitle:@"Color Image Item Tapped"];
  }];
  
  UIMenuItem *nextItem = [[UIMenuItem alloc] mik_initWithTitle:@"Show More Items..." action:^(UIMenuItem *item) {
    MenuItemAction action = ^(UIMenuItem *item) { [_self showAlertWithTitle:[item.title stringByAppendingString:@" Tapped"]]; };
    UIMenuItem *item1 = [[UIMenuItem alloc] mik_initWithTitle:@"1" action:action];
    UIMenuItem *item2 = [[UIMenuItem alloc] mik_initWithTitle:@"2" action:action];
    UIMenuItem *item3 = [[UIMenuItem alloc] mik_initWithTitle:@"3" action:action];
    controller.menuItems = @[item1, item2, item3];
    [controller setMenuVisible:YES animated:YES];
  }];
  
  controller.menuItems = @[textItem, imageItem, colorImageItem, nextItem];
  [controller setTargetRect:self.button.bounds inView:self.button];
  [controller setMenuVisible:YES animated:YES];
}

- (void)showAlertWithTitle:(NSString *)title
{
  UIAlertController *alertVC = [UIAlertController alertControllerWithTitle:title message:nil preferredStyle:UIAlertControllerStyleAlert];
  UIAlertAction *action = [UIAlertAction actionWithTitle:@"Dismiss" style:UIAlertActionStyleCancel handler:^(UIAlertAction *action) {}];
  [alertVC addAction:action];
  [self presentViewController:alertVC animated:YES completion:nil];
}

@end

File: Demo/Demo/ViewController.swift
//
//  ViewController.swift
//  Demo
//
//  Created by CHEN Xian’an on 1/16/16.
//  Copyright © 2016 lazyapps. All rights reserved.
//

import UIKit
import MenuItemKit

class ViewController: UIViewController {

  @IBOutlet var button: UIButton!
  
  override func viewDidLoad() {
    super.viewDidLoad()
    button.addTarget(self, action: #selector(self.tapButton(_:)), for: .touchUpInside)
  }

  @objc func tapButton(_ sender: AnyObject?) {
    let controller = UIMenuController.shared
    let textItem = UIMenuItem(title: "Text") { [weak self] _ in
      self?.showAlertWithTitle("text item tapped")
    }
    
    let image = UIImage(named: "Image")
    let imageItem = UIMenuItem(title: "Image", image: image) { [weak self] _ in
      self?.showAlertWithTitle("image item tapped")
    }

    let colorImage = UIImage(named: "ColorImage")
    let colorImageItem = UIMenuItem(title: "Image", image: colorImage) { [weak self] _ in
      self?.showAlertWithTitle("color image item tapped")
    }
    
    let nextItem = UIMenuItem(title: "Show More Items...") { _ in
      let action: MenuItemAction = { [weak self] in self?.showAlertWithTitle($0.title + " tapped") }
      let item1 = UIMenuItem(title: "1", action: action)
      let item2 = UIMenuItem(title: "2", action: action)
      let item3 = UIMenuItem(title: "3", action: action)
      controller.menuItems = [item1, item2, item3]
      controller.setMenuVisible(true, animated: true)
    }
    
    controller.menuItems = [textItem, imageItem, colorImageItem, nextItem]
    controller.setTargetRect(button.bounds, in: button)
    controller.setMenuVisible(true, animated: true)
  }

  func showAlertWithTitle(_ title: String) {
    let alertVC = UIAlertController(title: title, message: nil, preferredStyle: .alert)
    alertVC.addAction(UIAlertAction(title: "Dismiss", style: .cancel, handler: { _ in }))
    present(alertVC, animated: true, completion: nil)
  }

}
